src/main/java/io/cdap/plugin/http/common/pagination/BaseHttpPaginationIterator.java

```diff
@@ -84,13 +84,20 @@ protected boolean visitPageAndCheckStatusCode() throws IOException {
     if (response != null) { // close previous response
       response.close();
     }
+    if (this.shouldRetry()) {
+      getHttpClient().close(); // In case of retry attempt, close the previous HttpClient instance
```

Comment:
Could this give a null pointer exception, in case the API isn't able to create an http client

Reply:
getHttpClient returns the httpClient instance, which is created in the constructor and is never set to null. Hence I believe this can not be null at this point.